Add tool sdk to release pipeline and local pkg references

FedericoAmura opened the pull request:
# Description

This PR:
1. Adds vincent-tool-sdk to the sdk (and more in main) release pipeline. First publishing was done already [here](https://www.npmjs.com/package/@lit-protocol/vincent-tool-sdk)

2. Fixes the incorrect bundling directory of `@lit-protocol/vincent-sdk`. Check that vincent sdk had [its files](https://www.npmjs.com/package/@lit-protocol/vincent-sdk?activeTab=code) in `/dist/src`, duplicating README.md and package.json. Now it will publish without the extra `dist` as the [tool sdk](https://www.npmjs.com/package/@lit-protocol/vincent-tool-sdk?activeTab=code) or the [mcp](https://www.npmjs.com/package/@lit-protocol/vincent-mcp?activeTab=code)

3. Changes the references between our pkgs (dashboard including sdk, mcp including mcp, etc.) that we had hardcoded in a version with `workspace:*`

MaximusHaximus commented:
@FedericoAmura looks like the local reference isn't working from app-dashboard to the vincent-sdk (build failed with type errors about importing `jwt` from the vincent-sdk)

MaximusHaximus reviewed (changes requested):
It looks like the local workspace reference isn't working from app-dashboard to the vincent-sdk (build failed with type errors about importing j

FedericoAmura commented:
> It looks like the local workspace reference isn't working from app-dashboard to the vincent-sdk (build failed with type errors about importing j

Issue was an import from the `did-jwt` lib really. It wasn't fully compatible when building in another project. Changed and it worked
I also fixed the `mcp` build (importing the sdk required "dom" ts lib or it would crash when code has a `window` and a few other pipeline issues we had
The only things pending are the tools & policies build and test actions but I believe those are being worked on

MaximusHaximus reviewed (approved):
:+1: let's get this merged!

MaximusHaximus commented:
Rebased this on to main to resolve pnpm lock conflict -- merging this now that it's gtg, so that my tools and vincent-sdk PR can be based on these changes
